libc/utils/MPFRWrapper/MPFRUtils.cpp

```diff
@@ -351,6 +351,12 @@ class MPFRNumber {
     return result;
   }
 
+  MPFRNumber roundeven() const {
+    MPFRNumber result(*this);
+    mpfr_roundeven(result.value, value);
```

Comment:
Can you double check which version of MPFR adding `mpfr_roundeven` function?

Reply:
Can you add the version guard, and implement a replacement for `mpfr_roundeven` when it's not available, similar to https://github.com/llvm/llvm-project/blob/843cc474faefad1d639f4c44c1cf3ad7dbda76c8/libc/utils/MPFRWrapper/MPFRUtils.cpp#L244

Reply:
I implemented the replacement, but I am not sure if I should simply write `MPFR_VERSION_MAJOR >= 4`, or also specify the minor version. I went with the latter, but will change if required.

Reply:
Just check `MPFR_VERSION_MAJOR >= 4` should be enough.

Reply:
I also factored out the common code from the version guard.